### DISCUSSION TITLE
How can i change log level of artemis broker?

### DISCUSSION THREAD

**Discussion by lnthai2002** [arkmq-org/arkmq-org-broker-operator#891] [Q&A] [answered]
Hello, I followed this https://artemiscloud.io/docs/help/operator/#configuring-logging-for-brokers and to createda secret with the log config:
'''
apiVersion: v1
data:
  logging.properties: bW9uaXRvckludGVydmFsPTUgCnJvb3RMb2dnZXIubGV2ZWw9REVCVUcgCnJvb3RMb2dnZXIuYXBwZW5kZXJSZWYuY29uc29sZS5yZWY9Y29uc29sZSAKbG9nZ2VyLmFjdGl2ZW1xLm5hbWU9b3JnLmFwYWNoZS5hY3RpdmVtcSAKbG9nZ2VyLmFjdGl2ZW1xLmxldmVsPURFQlVHIApsb2dnZXIuYXJ0ZW1pc19zZXJ2ZXIubmFtZT1vcmcuYXBhY2hlLmFjdGl2ZW1xLmFydGVtaXMuY29yZS5zZXJ2ZXIgCmxvZ2dlci5hcnRlbWlzX3NlcnZlci5sZXZlbD1JTkZPIApsb2dnZXIuYXJ0ZW1pc19qb3VybmFsLm5hbWU9b3JnLmFwYWNoZS5hY3RpdmVtcS5hcnRlbWlzLmpvdXJuYWwgCmxvZ2dlci5hcnRlbWlzX2pvdXJuYWwubGV2ZWw9SU5GTyAKbG9nZ2VyLmFydGVtaXNfdXRpbHMubmFtZT1vcmcuYXBhY2hlLmFjdGl2ZW1xLmFydGVtaXMudXRpbHMgCmxvZ2dlci5hcnRlbWlzX3V0aWxzLmxldmVsPUlORk8gCmxvZ2dlci5jcml0aWNhbF9hbmFseXplci5uYW1lPW9yZy5hcGFjaGUuYWN0aXZlbXEuYXJ0ZW1pcy51dGlscy5jcml0aWNhbCAKbG9nZ2VyLmNyaXRpY2FsX2FuYWx5emVyLmxldmVsPUlORk8gCmxvZ2dlci5hdWRpdF9iYXNlLm5hbWU9b3JnLmFwYWNoZS5hY3RpdmVtcS5hdWRpdC5iYXNlIApsb2dnZXIuYXVkaXRfYmFzZS5sZXZlbD1PRkYgCmxvZ2dlci5hdWRpdF9iYXNlLmFkZGl0aXZpdHk9ZmFsc2UgCmxvZ2dlci5hdWRpdF9yZXNvdXJjZS5uYW1lPW9yZy5hcGFjaGUuYWN0aXZlbXEuYXVkaXQucmVzb3VyY2UgCmxvZ2dlci5hdWRpdF9yZXNvdXJjZS5sZXZlbD1PRkYgCmxvZ2dlci5hdWRpdF9yZXNvdXJjZS5hZGRpdGl2aXR5PWZhbHNlIApsb2dnZXIuYXVkaXRfbWVzc2FnZS5uYW1lPW9yZy5hcGFjaGUuYWN0aXZlbXEuYXVkaXQubWVzc2FnZSAKbG9nZ2VyLmF1ZGl0X21lc3NhZ2UubGV2ZWw9T0ZGIApsb2dnZXIuYXVkaXRfbWVzc2FnZS5hZGRpdGl2aXR5PWZhbHNlIApsb2dnZXIuamV0dHkubmFtZT1vcmcuZWNsaXBzZS5qZXR0eSAKbG9nZ2VyLmpldHR5LmxldmVsPVdBUk4gCmxvZ2dlci5hdXRoZW50aWNhdGlvbl9maWx0ZXIubmFtZT1pby5oYXd0LndlYi5hdXRoLkF1dGhlbnRpY2F0aW9uRmlsdGVyIApsb2dnZXIuYXV0aGVudGljYXRpb25fZmlsdGVyLmxldmVsPUVSUk9SIApsb2dnZXIuY3VyYXRvci5uYW1lPW9yZy5hcGFjaGUuY3VyYXRvciAKbG9nZ2VyLmN1cmF0b3IubGV2ZWw9V0FSTiAKbG9nZ2VyLnpvb2tlZXBlci5uYW1lPW9yZy5hcGFjaGUuem9va2VlcGVyIApsb2dnZXIuem9va2VlcGVyLmxldmVsPUVSUk9S
kind: Secret
metadata:
  creationTimestamp: "2024-04-22T16:29:47Z"
  name: artemis-logging-config
  namespace: ncp
  resourceVersion: "1163423477"
  uid: 89db25a0-ddea-4192-a4d3-b17453cffefa
type: Opaque
'''
However, when i checked the /amq-broker/etc/log4j2.properties i still see the default values for logs.
Am i missing something?

**Comment by brusdev** [comment]
@lnthai2002 your secret LGTM, it will be applied in place of `amq-broker/etc/log4j2.properties` but that file will not be replaced.

**Reply by lnthai2002** [reply]
Thank you for quick reply. However, using the config in the secret i shared above, I got the following error when the broker start:
```
-------------------------------------------------------------------
GMS: address=activemq-artemis-broker-ss-0-47659, cluster=activemq_broadcast_channel, physical address=10.0.15.78:7800
-------------------------------------------------------------------
ERROR StatusConsoleListener Unable to locate appender "console " for logger config "root"
ERROR StatusConsoleListener Unable to locate plugin type for Console
ERROR StatusConsoleListener Unable to locate plugin for PatternLayout
ERROR StatusConsoleListener Unable to locate plugin for Console
ERROR StatusConsoleListener Unable to invoke factory method in class org.apache.logging.log4j.core.config.AppendersPlugin for element Appenders: java.lang.NullPointerException: Cannot invoke "org.apache.logging.log4j.core.config.plugins.util.PluginType.getElementName()" because "childType" is null
 java.lang.NullPointerException: Cannot invoke "org.apache.logging.log4j.core.config.plugins.util.PluginType.getElementName()" because "childType" is null
        at org.apache.logging.log4j.core.config.plugins.visitors.PluginElementVisitor.visit(PluginElementVisitor.java:52)
        at org.apache.logging.log4j.core.config.plugins.util.PluginBuilder.generateParameters(PluginBuilder.java:286)
        at org.apache.logging.log4j.core.config.plugins.util.PluginBuilder.build(PluginBuilder.java:137)
        at org.apache.logging.log4j.core.config.AbstractConfiguration.createPluginObject(AbstractConfiguration.java:1138)
        at org.apache.logging.log4j.core.config.AbstractConfiguration.createConfiguration(AbstractConfiguration.java:1063)
        at org.apache.logging.log4j.core.config.AbstractConfiguration.doConfigure(AbstractConfiguration.java:664)
        at org.apache.logging.log4j.core.config.AbstractConfiguration.initialize(AbstractConfiguration.java:258)
        at org.apache.logging.log4j.core.config.AbstractConfiguration.start(AbstractConfiguration.java:304)
        at org.apache.logging.log4j.core.LoggerContext.setConfiguration(LoggerContext.java:621)
        at org.apache.logging.log4j.core.LoggerContext.onChange(LoggerContext.java:757)
        at org.apache.logging.log4j.core.util.AbstractWatcher$ReconfigurationRunnable.run(AbstractWatcher.java:93)
        at java.base/java.lang.Thread.run(Thread.java:840)
ERROR StatusConsoleListener Unable to locate appender "console " for logger config "root"
org.apache.activemq.artemis.component.WebTmpCleaner:NOTE: Picked up JDK_JAVA_OPTIONS: -Dbroker.properties=/amq/extra/secrets/activemq-artemis-broker-props/broker.properties
rpc error: code = NotFound desc = an error occurred when try to find container "38f3feefb79a5b00855066b164dc72e63c0135aca95ed7324d6c6e70e92e89da": not found
```

**Reply by brusdev** [reply, marked as answer]
@lnthai2002 your logging.properties is wrong:
```
monitorInterval=5 
rootLogger.level=DEBUG 
rootLogger.appenderRef.console.ref=console 
logger.activemq.name=org.apache.activemq 
logger.activemq.level=DEBUG 
logger.artemis_server.name=org.apache.activemq.artemis.core.server 
logger.artemis_server.level=INFO 
logger.artemis_journal.name=org.apache.activemq.artemis.journal 
logger.artemis_journal.level=INFO 
logger.artemis_utils.name=org.apache.activemq.artemis.utils 
logger.artemis_utils.level=INFO 
logger.critical_analyzer.name=org.apache.activemq.artemis.utils.critical 
logger.critical_analyzer.level=INFO 
logger.audit_base.name=org.apache.activemq.audit.base 
logger.audit_base.level=OFF 
logger.audit_base.additivity=false 
logger.audit_resource.name=org.apache.activemq.audit.resource 
logger.audit_resource.level=OFF 
logger.audit_resource.additivity=false 
logger.audit_message.name=org.apache.activemq.audit.message 
logger.audit_message.level=OFF 
logger.audit_message.additivity=false 
logger.jetty.name=org.eclipse.jetty 
logger.jetty.level=WARN 
logger.authentication_filter.name=io.hawt.web.auth.AuthenticationFilter 
logger.authentication_filter.level=ERROR 
logger.curator.name=org.apache.curator 
logger.curator.level=WARN 
logger.zookeeper.name=org.apache.zookeeper 
logger.zookeeper.level=ERROR
```

All lines end with a whitespace and it doesn't include the definition of the console appender, i.e.
```
# Console appender
appender.console.type=Console
appender.console.name=console
appender.console.layout.type=PatternLayout
appender.console.layout.pattern=%d %-5level [%logger] %msg%n
```

**Reply by lnthai2002** [reply]
good catch, thanks.
